integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java

```diff
@@ -297,10 +302,53 @@ private static boolean checkIfDisabled() {
     return Boolean.parseBoolean(isDisabled);
   }
 
+  private void commonConfigParse(ArgumentParser.ArgumentParserBuilder builder, SparkConf conf) {
+    builder
+        .timeout(findSparkConfigKeyDouble(conf, SPARK_CONF_TIMEOUT))
+        .apiKey(findSparkConfigKey(conf, SPARK_CONF_API_KEY).filter(str -> !str.isEmpty()))
+        .appName(findSparkConfigKey(conf, SPARK_CONF_APP_NAME).filter(str -> !str.isEmpty()))
+        .urlParams(findSparkUrlParams(conf, SPARK_CONF_URL_PARAM_PREFIX));
+
+    findSparkConfigKey(conf, SPARK_CONF_HOST_KEY).ifPresent(builder::host);
+    findSparkConfigKey(conf, SPARK_CONF_API_VERSION_KEY).ifPresent(builder::version);
+    findSparkConfigKey(conf, SPARK_CONF_NAMESPACE_KEY).ifPresent(builder::namespace);
+    findSparkConfigKey(conf, SPARK_CONF_JOB_NAME_KEY).ifPresent(builder::jobName);
+    findSparkConfigKey(conf, SPARK_CONF_PARENT_RUN_ID_KEY).ifPresent(builder::parentRunId);
+  }
+
   private ArgumentParser parseConf(SparkConf conf) {
     Optional<String> url = findSparkConfigKey(conf, SPARK_CONF_URL_KEY);
+    Optional<String> transportType = findSparkConfigKey(conf, SPARK_CONF_TRANSPORT_TYPE);
     if (url.isPresent()) {
       return ArgumentParser.parse(url.get());
+    } else if (transportType.isPresent()) {
+      // go and check the specific transport type setting
+      String mode = transportType.get();
+      Optional<TransportConfig> transportConfig = Optional.empty();
+      switch (mode) {
+        case "Kinesis":
```

Comment:
Would it make sense to do lower-case when comparing ?

Reply:
sure